src/repart/repart.c

```diff
@@ -7452,7 +7472,10 @@ static int context_fstab(Context *context) {
         if (r < 0)
                 return log_error_errno(r, "Failed to open temporary file for %s: %m", path);
 
-        fprintf(f, "# Automatically generated by systemd-repart\n\n");
+        if (arg_append_fstab == APPEND_AUTO)
+                fputs(AUTOMATIC_FSTAB_HEADER_START "\n", f);
+        else
+                fputs("# Automatically generated by systemd-repart\n\n", f);
```

Comment:
i don't thing we should generate different markers if "auto" is used or not. Always use the same markers, i.e. the start/end pair.

Reply:
Ok. Now there is one single kind of header.